logistics_project/apps/malawi/warehouse/runner.py

```diff
@@ -501,34 +495,57 @@ def update_consumption_times(since):
             c.time_needing_data = c.time_with_data
             c.save()
 
+
 def update_historical_data():
     """
     If we don't have a record of this supply point being updated, run
     through all historical data and just fill in with zeros.
     """
-    start_date = ReportingRate.objects.order_by('date')[0].date
-    warehouse_classes = [
+    hsa_level_start_date = ReportingRate.objects.filter(base_level=BaseLevel.HSA).order_by('date')[0].date
+    if settings.ENABLE_FACILITY_WORKFLOWS:
+        facility_level_start_date = ReportingRate.objects.filter(base_level=BaseLevel.FACILITY).order_by('date')[0].date
+
+    # These models are used by both base levels and have a product attribute
+    warehouse_classes_with_product = [
         ProductAvailabilityData,
+        CalculatedConsumption,
+        HistoricalStock,
+    ]
+
+    # These models are used by both base levels and have a base_level attribute
+    warehouse_classes_with_base_level = [
         ProductAvailabilityDataSummary,
         ReportingRate,
+    ]
+
+    # These models are used by only the HSA base level and have a product attribute
+    hsa_only_warehouse_classes_with_product = [
         OrderRequest,
         OrderFulfillment,
-        CalculatedConsumption,
-        HistoricalStock,
     ]
+
     print 'updating historical data'
-    for sp in SupplyPoint.objects.all():
-        try:
-            SupplyPointWarehouseRecord.objects.get(supply_point=sp)
-        except ObjectDoesNotExist:
-            # we didn't have a record so go through and historically update
-            # anything we maybe haven't touched
-            for year, month in months_between(start_date, sp.created_at):
+    for sp in SupplyPoint.objects.filter(supplypointwarehouserecord__isnull=True):
+        for year, month in months_between(hsa_level_start_date, sp.created_at):
+            window_date = datetime(year, month, 1)
+
+            for cls in (warehouse_classes_with_product + hsa_only_warehouse_classes_with_product):
+                _init_with_product(cls, sp, window_date, BaseLevel.HSA)
```

Comment:
for the models that don't have a `base_level` field are you just relying on / expecting that parameter to be ignored in the `get_or_create` call the function makes?